### PR TITLE
Adding Research Code from DPR & DARQ

### PR DESCRIPTION
**Pull request by tli2**
Prototype implementation for DPR and DARQ work.

### CODE REVIEW

File: cs/remote/src/FASTER.common/MessageType.cs [hunk 1]
```diff
@@ -71,5 +71,25 @@ public enum MessageType : byte
         /// Pending result
         /// </summary>
         PendingResult,
+        
+        /// <summary>
+        /// DARQ Enqueue
+        /// </summary>
+        DarqEnqueue,
```
Comment (badrishc) [inline cs/remote/src/FASTER.common/MessageType.cs:78]: DARQ specifics leaking into core/common

File: cs/remote/src/FASTER.common/SimpleObjectPool.cs [hunk 2]
```diff
@@ -12,20 +11,26 @@ namespace FASTER.common
     /// Object pool
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public class SimpleObjectPool<T> : IDisposable where T : class, IDisposable
+    public class SimpleObjectPool<T> : IDisposable where T : class
     {
         private readonly Func<T> factory;
+        private readonly Action<T> destructor;
```
Comment (badrishc) [inline cs/remote/src/FASTER.common/SimpleObjectPool.cs:17]: why this change? there is other code we have that depends on this.

File: cs/remote/src/FASTER.common/WireFormat.cs [hunk 3]
```diff
@@ -23,6 +23,10 @@ public enum WireFormat : byte
         /// Similar to DefaultVarLenKV but with WebSocket headers (binary)
         /// </summary>
         WebSocket = 2,
+        
+        DarqSubscribe,
```
Comment (badrishc) [inline cs/remote/src/FASTER.common/WireFormat.cs:27]: are these really new wire format level items? and DARQ doesnt seem like somthing to be exposed in this level.

File: cs/remote/src/FASTER.common/NetworkSenderBase.cs [hunk 4]
```diff
@@ -56,10 +58,7 @@ public NetworkSenderBase(int serverBufferSize)
         public abstract bool SendResponse(int offset, int size);
 
         /// <inheritdoc />
-        public abstract void SendResponse(byte[] buffer, int offset, int count, object context);
```
Comment (badrishc) [inline cs/remote/src/FASTER.common/NetworkSenderBase.cs:59]: i believe these abstracts are needed as they are used by other codes not in this repo.

File: cs/remote/src/FASTER.common/SimpleObjectPool.cs (hunk 2 above)
Comment (tli2) [reply cs/remote/src/FASTER.common/SimpleObjectPool.cs:17]: Needed to accommodate objects (e.g., SEAAs) that needs custom Dispose() logic. I have changed the default argument ordering so it should now be a backward compatible API change for existing code.

File: cs/remote/src/FASTER.common/NetworkSenderBase.cs (hunk 4 above)
Comment (tli2) [reply cs/remote/src/FASTER.common/NetworkSenderBase.cs:59]: Reverted

File: cs/remote/src/FASTER.common/WireFormat.cs (hunk 3 above)
Comment (tli2) [reply cs/remote/src/FASTER.common/WireFormat.cs:27]: Same as above for MessageType
